src/main/java/org/terasology/logic/health/ChangeMaxHealthAuthoritySystem.java

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.logic.health;
+
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.health.event.ChangeMaxHealthEvent;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.NUIManager;
+import org.terasology.rendering.nui.widgets.UIIconBar;
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class ChangeMaxHealthAuthoritySystem extends BaseComponentSystem {
+    @In
+    NUIManager nuiManager;
+
+    @ReceiveEvent
+    public void changeMaxHealth(ChangeMaxHealthEvent event, EntityRef player, HealthComponent health) {
```

Comment:
this should be the last receiver of the event, so that no other entity can affect the max health after we updated the bar, so we should
- have the lowest priority
- consume the event

Reply:
AbstractValueModifiableEvent is not consumable though...  But, Changed the priority.